backend/btrixcloud/basecrawls.py

```diff
@@ -190,28 +190,33 @@ async def update_crawl(
         return {"updated": True}
 
     async def delete_crawls(
-        self, org: Organization, delete_list: DeleteCrawlList, type_=None
+        self, org: Organization, delete_list: DeleteCrawlList, type_: str
     ):
         """Delete a list of crawls by id for given org"""
-        cids_to_update = set()
+        cids_to_update: dict[str, dict[str, int]] = {}
 
         size = 0
 
         for crawl_id in delete_list.crawl_ids:
             crawl = await self.get_crawl_raw(crawl_id, org)
-            size += await self._delete_crawl_files(crawl, org)
```

Comment:
Can filter by crawl type here, since we have the type available. Files are deleted for all crawls, but then can do:
```
deleted_size = await self._delete_crawl_files(crawl, org)
if crawl.type == "crawl":
   crawl_size += delete_size
else:
   upload_size += delete_size
```

Reply:
Actually, it looks like we're not even checking the type before deleting crawl files here!
Should probably do:
```
if type_ and crawl.type != type_:
   continue
```

This is actually a bug in the current version as well, where it'll delete the files regardless of type, but not actual archived item object!

Reply:
Good catch on the missing type check! That's now been added.

I think there are advantages to handling the complexity of deleting multiple types in `delete_crawls_all_types` rather than `delete_crawls`. Everywhere else in our app that we are deleting content other than through the `/all-crawls` delete endpoint, we're handling one type at a time. And having `delete_crawls` handle only one type at a time keeps things a bit simpler, e.g. letting `inc_org_bytes_stored`only need to worry about one archived item type at a time.

I pushed a change to `delete_crawls_all_types` to split the delete list into crawls and uploads arrays in one pass rather than two, which should help.

There are other improvements to be made to the delete endpoints that I'm planning on doing in a separate pass in this sprint (see https://github.com/webrecorder/browsertrix-cloud/issues/1208), so I'd expect some of this to change and get optimized there, but I don't want to hold up this PR for something that's creeping a bit out of scope just because I discovered some bugs 😅 If that sounds reasonable to you!